packages/lix-sdk/src/branch/merge-branch.ts

```diff
@@ -0,0 +1,171 @@
+
+		// let plugin detect conflicts
+		await Promise.all(
+			plugins.map(async (plugin) => {
+				if (plugin.detectConflictsV2) {
+					const conflicts = await plugin.detectConflictsV2({
+						lix: args.lix,
+						changes: diffingChanges,
+					});
+					detectedConflicts.push(...conflicts);
+				}
+			}),
+		);
+
+		// update the branch change pointers for non-conflicting changes
+		await Promise.all(
+			diffingPointers.map(async (pointer) => {
+				// potentially expensive query
+				const pluginDetectedConflict = detectedConflicts.find(
+					(conflict) =>
+						conflict.change_id === pointer.target_change_id ||
+						conflict.change_id === pointer.source_change_id,
+				);
+				if (pluginDetectedConflict) {
+					// don't update the branch change pointer
+					return;
+				}
+
+				// if the entity change doesn't exist in the target
+				// it can't conflict (except if a plugin detected
+				// a semantic conflict)
+				const hasConflictingEntityChanges = !pointer.target_change_id
+					? false
+					: await childOfCommonAncestorDiffers({
+							lix: { db: trx },
+							changeA: { id: pointer.source_change_id },
+							changeB: { id: pointer.target_change_id },
+						});
+
+				if (hasConflictingEntityChanges) {
+					detectedConflicts.push({
+						change_id: pointer.source_change_id,
+						conflicting_change_id: pointer.target_change_id!,
+					});
+					return;
+				}
+
+				await trx
+					.insertInto("branch_change_pointer")
+					.values([
+						{
+							branch_id: args.targetBranch.id,
+							change_id: pointer.source_change_id,
+							change_file_id: pointer.change_file_id,
+							change_entity_id: pointer.change_entity_id,
+							change_type: pointer.change_type,
+						},
+					])
+					.onConflict((oc) =>
+						oc.doUpdateSet({ branch_id: args.targetBranch.id }),
+					)
+					.execute();
+			}),
+		);
+
+		// insert the detected conflicts
+		// (ignore if the conflict already exists)
+		if (detectedConflicts.length > 0) {
+			await trx
+				.insertInto("conflict")
+				.values(detectedConflicts)
+				.onConflict((oc) => oc.doNothing())
+				.execute();
+		}
+	};
+
+	if (args.lix.db.isTransaction) {
+		return executeInTransaction(args.lix.db);
+	} else {
+		return args.lix.db.transaction().execute(executeInTransaction);
+	}
+}
+
+// could be optimized by returning the diffing changes
+async function getBranchChangePointerDiff(
+	trx: Lix["db"],
+	args: {
+		sourceBranch: Branch;
+		targetBranch: Branch;
+	},
+) {
+	return await trx
+		// select all change pointers from the source branch
+		.selectFrom("branch_change_pointer as source")
+		.where("source.branch_id", "=", args.sourceBranch.id)
+		// and join them with the change pointers from the target branch
+		.leftJoin("branch_change_pointer as target", (join) =>
+			join
+				// join related change pointers
+				// related = same entity, file, and type
+				.onRef("source.change_entity_id", "=", "target.change_entity_id")
```

Comment:
ok - this seems to be the reason for the extra columns in branch_change_pointer

Reply:
Yes. 

- (minor) avoid joining pointers with the change table 
- (major) ensure consistency that a branch only points to one "entity change" at a time